### ISSUE TITLE
Ghost clipping

### ISSUE THREAD

**Issue by ffuuugor** [pytorch/opacus#349]
Paper by [Li et al](https://arxiv.org/pdf/2110.05679.pdf) demonstrated successful fine-tuning of GPT-2 with DP and proposed multiple improvements to make the training more efficient.
One of particular interest to Opacus is ghost clipping, which doesn't require materializing per sample gradients, which can lead to significant memory wins.
As far as I understand, it comes at a certain gpu processing cost, so we first need to evaluate if it's beneficial at least in some cases, and if yes - implement this as an option in opacus

**Comment by lxuechen** [comment]
Just a note that things are implemented [here](https://github.com/lxuechen/private-transformers) in case you need a reference. 

I can help out if you decide that this may be helpful for the privacy research happening in your team.

**Comment by iden-kalemaj** [comment]
Closing this issue as ghost clipping and fast gradient clipping are now implemented in Opacus.
